src/v5/core/action.ts

```diff
@@ -17,13 +17,16 @@ import { allowStateReadsStart, allowStateReadsEnd } from "./derivation"
 // mobx versions
 let currentActionId = 0
 let nextActionId = 1
+const isProduction = process.env.NODE_ENV === "production"
+const functionNameDescriptor = Object.getOwnPropertyDescriptor(() => {}, "name")
+const isFunctionNameConfigurable = functionNameDescriptor && functionNameDescriptor.configurable
 
 export interface IAction {
     isMobxAction: boolean
 }
 
 export function createAction(actionName: string, fn: Function, ref?: Object): Function & IAction {
-    if (process.env.NODE_ENV !== "production") {
+    if (!isProduction) {
```

Comment:
best keep the condition here, otherwise it kills the dead code elimination when making production builds

Reply:
Yea, eventually I want to use TSDX here as well so we could stick to a simple `__DEV__` condition instead of this wordy thing, but for now, it needs to be like this.

Reply:
@FredyC I make the changes, let me know if this is your intent.